extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -306,17 +323,33 @@ private function createSmartGroupItem( $rowItems, $rowSpan ) {
 	}
 
 	private function createSmartGroupSections( $rowItems, $capacity ) {
-		return array_reduce( $rowItems, function ( $result, $item ) use ( $capacity ) {
-			$styles = "width: calc({$item['data']['span']} / $capacity * 100%);";
-
-			$label = $item['data']['label'] ?? "";
-			if ( !empty( $label ) ) {
-				$result['renderLabels'] = true;
-			}
-			$result['labels'][] = [ 'value' => $label, 'inlineStyles' => $styles ];
-			$result['values'][] = [ 'value' => $item['data']['value'], 'inlineStyles' => $styles ];
+		return array_reduce(
```

Comment:
It might be more readable to do this with a `foreach` loop, PHP's "functional" APIs are not too intuitive.

Reply:
actually I would leave it as is, since I'd need to modify data in array I'm looping over or create a second array which also would not be the prettiest solution